### ISSUE TITLE
Follow up issue to # 15048 - Server -side cookie expiration limit shows 2 days instead of 6 months

### ISSUE THREAD

**Issue by MadhaviSeelam** [brave/brave-browser#25787]
## Steps to Reproduce

   1. Install 1.45.84
   2. launch Brave
   3. visited https://fmarier.org/document-cookie/set-cookie4.html
   4. Opened the `Application` tab in the `devtools`.
   5. Clicked on `Storage`, then `Cookies `in the sidebar.
   6. Clicked on https://fmarier.org and looked at the expiry column.
   7. Opened `brave://settings/cookies/detail?site=fmarier.org` and checked the expiry date

## Actual result:

expiry date shows Oct 5, 2022 (2022-10-05T22:04:30.414Z) instead of 6 months  (i.e. April 01, 2023)

ex1|ex2
----|----
<img width="956" alt="image" src="https://user-images.githubusercontent.com/98358127/193694406-e12278c0-2078-4d4e-a9a9-38ba1f64bde9.png">|<img width="476" alt="image" src="https://user-images.githubusercontent.com/98358127/193694761-a4035ef4-8873-4f64-90cf-7559ae20f109.png">

## Expected result:
Should expire 6 months from the current date (i.e. 2023-04-01T22:09:13.740Z)

## Reproduces how often: 

<img width="954" alt="image" src="https://user-images.githubusercontent.com/98358127/193694972-ff18d51e-323d-4eba-b43b-3a26c05eb403.png">

## Brave version (brave://version info)

Brave | 1.45.84 Chromium: 106.0.5249.91 (Official Build) beta (64-bit)
-- | --
Revision | fa96d5f07b1177d1bf5009f647a5b8c629762157-refs/branch-heads/5249@{#707}
OS | Windows 11 Version 21H2 (Build 22000.978)

## Version/Channel Information:

- Can you reproduce this issue with the current release? N/A
- Can you reproduce this issue with the beta channel?  Yes
- Can you reproduce this issue with the nightly channel? Yes

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards? 
- Is the issue reproducible on the latest version of Chrome? 

## Miscellaneous Information:
 @ShivanKaul  @rebron  
cc: @stephendonner

**Comment by ShivanKaul** [comment]
@MadhaviSeelam the issue here is that the cookie expires in 2 days, right? I see that for https://fmarier.org/document-cookie/set-cookie4.html the server sends back header `set-cookie: test4=yes; max-age=172800` which is 2 days. Looks like the test plan in https://github.com/brave/brave-core/pull/1905#issue-419029258 is incorrect, and should say that for set-cookie4.html the expected days is 2 days, and not 6 months. I see that https://github.com/brave/brave-browser/issues/3443#issuecomment-490589789 also verified that set-cookie4.html sets cookie for 2 days. I'll update the test plan. 

TL;DR: The test plan was incorrect. Let me know how that sounds!

**Comment by MadhaviSeelam** [comment]
@ShivanKaul sounds good to me.  Will close this issue in that case.

**Comment by MadhaviSeelam** [comment]
Closing the issue as the testplan amended for the correct expiration date and as per above comment for the scenario.

**Comment by cemerson** [comment]
My apologies if I should post this as a new issue but it seems related (?). I'm setting a cookie to expire in 30 days but Brave (and Opera) show an expiration date of 7 days from now vs 30 (?). Is this just a visual bug or is the cookie actually setting the wrong date in Brave too?

![browser-cookie-expdate-bugs-20230317064240](https://user-images.githubusercontent.com/1080397/225884669-ae77fc97-ec94-426b-9239-87fadac34a00.png)
